### DISCUSSION TITLE
Converting SPICE dataset to extxyz (best practices)

### DISCUSSION THREAD

**Discussion by ERovers** [mir-group/allegro#34] [Q&A]
Hi!

I'm intrigued by the paper "Scaling the leading accuracy of deep equivariant models to biomolecular simulations of realistic size" and would like to try it for an application where I want to run MD on large biomolecular system (90K atoms). I'm more familiar with other MD simulation softwares like AMBER and GROMACS, so apologies if this is a very naive question. 

For training on the SPICE dataset, how would you convert the hdf5 database to the extxyz file? The dataset contains the atomic coordinates, forces and energy, so can imagine to create a file and parsing this information. But when I looked at the Silicon example notebook extxyz file there are also parameters like lattice (box?), stress and pbc. How would you get this information (with ASE?) or is it not necessary for training?

Thanks for any help!

**Comment by Linux-cpp-lisp** [comment]
Hi @ERovers ,

Thanks for your interest in our work!

The easiest way to create extXYZ files for Allegro / NequIP is using `ase.io.write`.  You only need to include properties like the box (lattice) or stress if you have them in your data; `nequip` will correctly not ask for them unless you do something inconsistent (such as include `stress` in your loss function without including it in your dataset). `pbc` just indicated whether the frame is periodic.

> is it not necessary for training?

In general, you can train in `nequip` on any combination of energy, forces, and stresses/virials (or other properties), but generally the minimal required is forces or forces and energies. It depends a lot on your system.

**Comment by vputz** [comment]
Somewhat late to this answer, but as it's something we were recently struggling with, here's a little command-line tool which I've written to extract whole subsets of the SPICE dataset to ext
xyz along with a "subset metrics" routine to list the atomic symbols and minimum required r_cutoff to ensure you include at least N neighbors (after exporting once and having r_cutoff too low, resulting in failure when some atoms had no neighbors within r_cutoff).  Requires a few libraries (ase, h5py, scipy, rich, typer), and attached as txt rather than py because github.

[export_spice.txt](https://github.com/mir-group/allegro/files/11956093/export_spice.txt)

Mind you, I'm not completely sure it does a completely _correct_ export, but we were able to do minimum training on the results, so there's that!

**Reply by Linux-cpp-lisp** [reply]
Thanks for sharing @vputz !
